lib/buffer.js

```diff
@@ -1021,557 +1002,786 @@ Buffer.prototype.slice = function slice(start, end) {
+      last);
+}
+
+function readUInt40BE(buf, offset) {
+  checkOffsetType(offset);
+  const first = buf[offset];
+  const last = buf[offset + 4];
+  if (first === undefined || last === undefined)
+    boundsError();
+
+  return first * SHIFT_LEFT_32 +
+    (buf[offset + 1] * 0x1000000) +
+    ((buf[offset + 2] << 16) |
+      (buf[offset + 3] << 8) |
+      last);
+}
 
-Buffer.prototype.readUInt32BE = function readUInt32BE(offset, noAssert) {
-  offset = offset >>> 0;
-  if (!noAssert)
-    checkOffset(offset, 4, this.length);
+Buffer.prototype.readUInt32BE = function readUInt32BE(offset) {
+  checkOffsetType(offset);
+  const first = this[offset];
+  const last = this[offset + 3];
+  if (first === undefined || last === undefined)
+    boundsError();
 
   return (this[offset] * 0x1000000) +
       ((this[offset + 1] << 16) |
       (this[offset + 2] << 8) |
       this[offset + 3]);
 };
 
+function readUInt24BE(buf, offset) {
+  checkOffsetType(offset);
+  const first = buf[offset];
+  const last = buf[offset + 2];
+  if (first === undefined || last === undefined)
+    boundsError();
 
-Buffer.prototype.readIntLE = function readIntLE(offset, byteLength, noAssert) {
-  offset = offset >>> 0;
-  byteLength = byteLength >>> 0;
-
-  if (!noAssert) {
-    checkByteLength(byteLength);
-    checkOffset(offset, byteLength, this.length);
-  }
-
-  var val = this[offset];
-  var mul = 1;
-  var i = 0;
-  while (++i < byteLength && (mul *= 0x100))
-    val += this[offset + i] * mul;
-  mul *= 0x80;
+  return (first << 16) | (buf[offset + 1] << 8) | last;
+}
 
-  if (val >= mul)
-    val -= Math.pow(2, 8 * byteLength);
+Buffer.prototype.readUInt16BE = function readUInt16BE(offset) {
+  checkOffsetType(offset);
+  const first = this[offset];
+  const last = this[offset + 1];
+  if (first === undefined || last === undefined)
+    boundsError();
 
-  return val;
+  return (this[offset] << 8) | this[offset + 1];
 };
 
+Buffer.prototype.readIntLE = function readIntLE(offset, byteLength) {
+  if (byteLength === 6)
+    return readInt48LE(this, offset);
+  if (byteLength === 5)
+    return readInt40LE(this, offset);
+  if (byteLength === 3)
+    return readInt24LE(this, offset);
+  if (byteLength === 4)
+    return this.readInt32LE(offset);
+  if (byteLength === 2)
+    return this.readInt16LE(offset);
+  if (byteLength === 1)
+    return this.readInt8(offset);
+
+  byteLengthError(byteLength);
+};
 
-Buffer.prototype.readIntBE = function readIntBE(offset, byteLength, noAssert) {
-  offset = offset >>> 0;
-  byteLength = byteLength >>> 0;
-
-  if (!noAssert) {
-    checkByteLength(byteLength);
-    checkOffset(offset, byteLength, this.length);
-  }
+function readInt48LE(buf, offset) {
+  checkOffsetType(offset);
+  const first = buf[offset];
+  const last = buf[offset + 5];
+  if (first === undefined || last === undefined)
+    boundsError();
+
+  const val = buf[offset + 4] | (last << 8);
+  return ((val & 0x8000) ? val | 0xffff0000 : val) * SHIFT_LEFT_32 +
```

Comment:
I tried this but either I did not understand the way you would like to do this or there is at least one value that this would not work for. After looking at it again, I changed the check a tiny bit and those are a bit faster now but it is not that significant.

Can you please provide me with a full example?